crates/solver/src/settlement_post_processing/optimize_score.rs

```diff
@@ -0,0 +1,49 @@
+use {
+    super::SettlementSimulating,
+    crate::{
+        objective_value::Inputs,
+        settlement::Settlement,
+        solver::score_computation::ScoreCalculator,
+    },
+    gas_estimation::GasPrice1559,
+    num::{BigRational, FromPrimitive},
+    shared::{external_prices::ExternalPrices, http_solver::model::Score},
+};
+
+pub async fn optimize_score(
+    settlement: Settlement,
+    settlement_simulator: &impl SettlementSimulating,
+    score_calculator: &ScoreCalculator,
+    gas_price: GasPrice1559,
+    prices: &ExternalPrices,
+) -> Settlement {
+    if settlement.score.is_some() {
+        return settlement;
+    }
+
+    let gas_amount = match settlement_simulator
+        .settlement_would_succeed(settlement.clone())
+        .await
+    {
+        Ok(gas_amount) => gas_amount * 9 / 10, /* multiply with 0.9 to get more realistic gas */
+        // amount
+        Err(_) => return settlement,
```

Comment:
This can only happen if some random error happened during the simulation or if a new block got mined since the last optimization step where the settlement no longer works.
In both cases I would err on the safe side and set the `score` to 0 here to avoid having to execute a reverting solution.

Reply:
I kept is as is. The reasoning is that this component is expected to either improve the solution or do nothing. We do have a simulation again in the ranking phase that would cover failure. Setting score to 0 here is one side effect that I would prefer to avoid if possible (it would mean that it is the value provided by solver and harder to debug).